google/cloud/spanner/internal/date.cc

```diff
@@ -13,42 +13,44 @@
 // limitations under the License.
 
 #include "google/cloud/spanner/internal/date.h"
-#include "google/cloud/spanner/internal/time_format.h"
-#include <ctime>
+#include <array>
+#include <cinttypes>
+#include <cstdio>
 
 namespace google {
 namespace cloud {
 namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
 namespace internal {
 
-namespace {
-
-// RFC3339 "full-date".
-constexpr auto kDateFormat = "%Y-%m-%d";
-
-}  // namespace
-
 std::string DateToString(Date d) {
-  std::tm tm;
-  tm.tm_year = static_cast<int>(d.year() - 1900);
-  tm.tm_mon = d.month() - 1;
-  tm.tm_mday = d.day();
-  return FormatTime(kDateFormat, tm);
+  std::array<char, sizeof "-9223372036854775807-01-01"> buf;
```

Comment:
Not that it matters, but is `-9223372036854775807` supposed to represent `INT64_MIN`? Shouldn't this be `-9223372036854775808` then? Alternatively, put a comment for what this magic number is supposed to mean.

Reply:
Yes, it should be.  Fixed.  Thanks!